polkadot/consensus/src/error.rs

```diff
@@ -16,45 +16,30 @@
 
 //! Errors that can occur during the consensus process.
 
-use primitives::block::{HeaderHash, Number};
+use polkadot_primitives::AccountId;
+
 error_chain! {
 	links {
 		PolkadotApi(::polkadot_api::Error, ::polkadot_api::ErrorKind);
 		Bft(::bft::Error, ::bft::ErrorKind);
 	}
 
-	foreign_links {
-		Io(::std::io::Error);
-		SharedIo(::futures::future::SharedError<::std::io::Error>);
-	}
-
 	errors {
 		InvalidDutyRosterLength(expected: usize, got: usize) {
 			description("Duty Roster had invalid length"),
 			display("Invalid duty roster length: expected {}, got {}", expected, got),
 		}
-		ProposalNotForPolkadot {
-			description("Proposal provided not a Polkadot block."),
-			display("Proposal provided not a Polkadot block."),
-		}
-		TimestampInFuture {
-			description("Proposal had timestamp too far in the future."),
-			display("Proposal had timestamp too far in the future."),
-		}
-		WrongParentHash(expected: HeaderHash, got: HeaderHash) {
-			description("Proposal had wrong parent hash."),
-			display("Proposal had wrong parent hash. Expected {:?}, got {:?}", expected, got),
+		NotValidator(id: AccountId) {
+			description("Local account ID not a validator at this block."),
+			display("Local account ID ({:?}) not a validator at this block.", id),
 		}
-		WrongNumber(expected: Number, got: Number) {
-			description("Proposal had wrong number."),
-			display("Proposal had wrong number. Expected {:?}, got {:?}", expected, got),
+		PrematureDestruction {
+			description("Proposer destroyed before finishing proposing or evaluating"),
+			display("Proposer destroyed before finishing proposing or evaluating"),
 		}
-		ProposalTooLarge(size: usize) {
-			description("Proposal exceeded the maximum size."),
-			display(
-				"Proposal exceeded the maximum size of {} by {} bytes.",
-				::MAX_TRANSACTIONS_SIZE, ::MAX_TRANSACTIONS_SIZE.saturating_sub(*size)
-			),
+		Timer(e: String) {
```

Comment:
has to be a string because we need to construct this if the shared delay fails. io::Error is not `Clone`